experimental/Schemes/IdealSheaves.jl

```diff
@@ -576,6 +578,164 @@ function primary_decomposition(I::Union{<:MPolyQuoIdeal, <:MPolyQuoLocalizedIdea
+end
+
+
+function minimal_associated_points(I::IdealSheaf)
+  X = scheme(I)
+  OOX = OO(X)
+
+  charts_todo = copy(affine_charts(X))            ## todo-list of charts
+
+  associated_primes_temp = Vector{IdDict{AbsSpec, Ideal}}()  ## already identified components
+                                                  ## may not yet contain all relevant charts. but
+                                                  ## at least one for each identified component
+
+# run through all charts and try to match the components
+  while length(charts_todo) > 0
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                        ## supp(I) might not meet all components
+    components_here = minimal_primes(I(U))
+
+## run through all primes in MinAss(I(U)) and try to match them with previously found ones
+    for comp in components_here
+      matches = match_on_intersections(X,U,comp,associated_primes_temp,false)
+      nmatches = length(matches)
+
+      if nmatches == 0                             ## not found
+        add_dict = IdDict{AbsSpec,Ideal}()         ## create new dict
+        add_dict[U] = comp                         ## and fill it
+        push!(associated_primes_temp, add_dict)
+      elseif nmatches == 1                         ## unique match, update it
+        component_index = matches[1]
+        associated_primes_temp[component_index][U] = comp
+      else                                                ## more than one match, form union
+        target_comp = pop!(matches)
+        merge!(associated_primes_temp[target_comp], associated_primes_temp[x] for x in matches)
+        deleteat!(associated_primes_temp,matches)
+        associated_primes_temp[target_comp][U] = comp
+      end
+    end
+  end
+
+# fill the gaps arising from a support not meeting a patch
+  for U in affine_charts(X)
+    I_one = ideal(OOX(U),one(OOX(U)))
+    for i in 1:length(associated_primes_temp)
+      !haskey(associated_primes_temp[i],U) || continue
+      associated_primes_temp[i][U] = I_one
+    end
+  end
+
+# make sure to return ideal sheaves, not dicts
+  associated_primes_result = [IdealSheaf(X,associated_primes_temp[i],check=false) for i in 1:length(associated_primes_temp)]
+  return associated_primes_result
+end
+
+function associated_points(I::IdealSheaf)
+  X = scheme(I)
+  OOX = OO(X)
+  charts_todo = copy(affine_charts(X))            ## todo-list of charts
+
+  associated_primes_temp = Vector{IdDict{AbsSpec, Ideal}}()  ## already identified components
+                                                  ## may not yet contain all relevant charts. but
+                                                  ## at least one for each identified component
+
+# run through all charts and try to match the components
+  while !iszero(length(charts_todo))
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                        ## supp(I) might not meet all components
+    components_here = [ a for (_,a) in primary_decomposition(I(U))]
```

Comment:
This looks like you are doing a primary decomposition in every chart. 
Naively I would have done the following:
1. Choose my favorite chart $U_1$
2. Do a primary decomposition of  $\mathcal{I}(U_1)$ 
3. Go to the next chart and saturate $\mathcal{I}(U_2)$ by the defining ideal of $U_1 \setminus U_2$
4. Compute a primary decomposition again. Hopefully this is faster after the saturation.
5. Continue like this....
Assemble the result (no primary decompositions and no matchings needed)
Is this too naive?

Reply:
(In any case this does not need to keep us from merging this PR)

Reply:
Actually, I am envisioning implementation of this approach for the 'primary decomposition' of the ideal sheaf -- this is just 'associated primes'. The potential drawback of the extending approach is that
- passage from chart to chart is expensive
- saturation by non-principal ideals is (very) expensive
- but so is primary decomposition as well
and hence this is not guaranteed to be faster or slower, but it depends on the specific ideal which approach is more efficient. Since we need the extension style approach for the 'primary decomposition' anyway, I will implement it there.